src/app/login-button/login-button.component.html

```diff
@@ -0,0 +1,21 @@
+<div *ngIf="auth.user$ | async; then showLogoutButton else showLoginButton">
+  <!-- template will replace this div -->
+</div>
+
+<!--Authenticated-->
+<ng-template #showLogoutButton>
+  <div *ngIf="auth.user$ | async as user">
+    <!-- Display User Name -->
+    <span>You're logged in as {{user.displayName}} </span>
+    <button mat-raised-button color="primary" id="logout_button" (click)="onLogoutClick()">
+      Logout
+    </button>
+  </div>
+</ng-template>
+
+<!--Not authenticated-->
+<ng-template #showLoginButton>
+    <button mat-raised-button color="primary" id="login_button" (click)="onLoginClick()">
```

Comment:
Why don't you just use a link instead of a button?

Reply:
What do you mean?